### PR TITLE
add tags package

### PR DESCRIPTION
**Pull request by balamurugana**
common tags package for server, minio-go and mc.

### CODE REVIEW

File: pkg/tags/tags.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+		}
+	}
+
+	return tagging, nil
+}
+
+// UnmarshalXML decodes XML data of tags in reader specified in
+// https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketTagging.html#API_PutBucketTagging_RequestSyntax.
+// If isObject is set, it validates for object tags.
+func UnmarshalXML(reader io.Reader, isObject bool) (*Tags, error) {
```
Comment (harshavardhana) [inline pkg/tags/tags.go:279]: Would be more readable to call these ParseObject and ParseBucket without taking the boolean for a public API, you may use this internally.

File: pkg/tags/tags.go [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+	if err := xml.NewDecoder(reader).Decode(tagging); err != nil {
+		return nil, err
+	}
+
+	return tagging, nil
+}
+
+// Parse decodes HTTP query formatted string into tags. A query formatted string is like "key1=value1&key2=value2".
+func Parse(s string, isObject bool) (*Tags, error) {
```
Comment (harshavardhana) [inline pkg/tags/tags.go:295]: Bucket tagging is not supported with query formatted string, looks like isObject is redundant here.

File: pkg/tags/tags.go (hunk 1 above)
Comment (balamurugana) [reply pkg/tags/tags.go:279]: I am doing with slight change in names as `ParseBucketXML()` and `ParseObjectXML()`

File: pkg/tags/tags.go (hunk 2 above)
Comment (balamurugana) [reply pkg/tags/tags.go:295]: Going to change the name as `ParseObjectTags()` without `isObject bool`

File: pkg/tags/tags.go [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions
+const (
+	maxKeyLength      = 128
+	maxValueLength    = 256
+	maxObjectTagCount = 10
+	maxTagCount       = 50
+)
+
+func checkKey(key string) error {
+	if len(key) == 0 || utf8.RuneCountInString(key) > maxKeyLength || strings.Contains(key, "&") {
```
[the remark this answers is not in the document]
Comment (balamurugana) [reply pkg/tags/tags.go:67]: They are allowed. No change I made in this PR with previous checks. If we want stricter checks, let me know.

[the remark this answers is not in the document]
Comment (kannappanr) [reply pkg/tags/tags.go:67]: I think we can take this PR, but we probably should add stricter checks

[the remark this answers is not in the document]
Comment (harshavardhana) [reply pkg/tags/tags.go:67]: :+1:

**Pull request merged**